typedspark/_utils/load_table.py

```diff
@@ -15,14 +16,26 @@
 from typedspark._utils.register_schema_to_dataset import register_schema_to_dataset
 
 
+def _replace_illegal_column_names(dataframe: DataFrame) -> DataFrame:
+    """Replaces illegal column names with a legal version."""
+    for column in dataframe.columns:
+        dataframe = dataframe.withColumnRenamed(column, _replace_illegal_characters(column))
+    return dataframe
+
+
+def _replace_illegal_characters(column_name: str) -> str:
+    """Replaces illegal characters in a column name with an underscore."""
+    return re.sub("[^A-Za-z0-9]", "_", column_name)
```

Comment:
Could automatically replacing the illegal characters cause undesired behavior? When the user thinks they are creating a dataframe with 2 distinct columns with something like:
```
a!
a&
```
which both get converted into `a_`

Reply:
Good point! I've edited the function to raise a ValueError when this happens.